scripts/build.js

```diff
@@ -156,6 +153,34 @@ function build(options) {
   });
 }
 
+function resolveNodeGyp(options) {
+  var args = ['rebuild', '--verbose'].concat(
+    ['libsass_ext', 'libsass_cflags', 'libsass_ldflags', 'libsass_library'].map(function (subject) {
+      return ['--', subject, '=', process.env[subject.toUpperCase()] || ''].join('');
+    })).concat(options.args);
+
+  // For node-chakracore, check if node-gyp is in the path.
+  // If yes, use it instead of using node-gyp directly from
+  // node_modules because the one in node_modules is not
```

Comment:
Since it looks like the only difference (by just browsing the History) for your copy of gyp looks like https://github.com/nodejs/node-chakracore/commit/023b96c446c90f75b2e98a6e6240f935ff7b8375 and https://github.com/nodejs/node-chakracore/commit/3a09525dd257357a3a763d4a0451daa1b3678c2c. Have you tried to just upstream those changes to gyp? Then it would just be a matter of bumping the minimum node-gyp version